src/github.com/matrix-org/dendrite/clientapi/readers/profile.go

```diff
@@ -159,3 +196,49 @@ func getLocalPart(userID string) string {
 	// Return the part after the "@"
 	return strings.Split(username, "@")[1]
 }
+
+// Send an update using kafka to notify the roomserver of the profile update
+// Returns an error if the update failed to send
+func sendUpdate(userID string, oldProfile *authtypes.Profile, newAvatarURL string,
+	newDisplayName string, accountDB *accounts.Database, cfg config.Dendrite,
+) error {
+	var update profileUpdate
+	var m sarama.ProducerMessage
+
+	m.Topic = string(cfg.Kafka.Topics.UserUpdates)
+	m.Key = sarama.StringEncoder(userID)
+
+	// Determining if the changed value is the avatar URL or the display name
+	if len(newAvatarURL) > 0 {
+		update = profileUpdate{
+			Updated:  "avatar_url",
+			OldValue: oldProfile.AvatarURL,
+			NewValue: newAvatarURL,
+		}
+	} else if len(newDisplayName) > 0 {
+		update = profileUpdate{
+			Updated:  "displayname",
+			OldValue: oldProfile.DisplayName,
+			NewValue: newDisplayName,
+		}
+	} else {
+		panic(fmt.Errorf("No update to send"))
+	}
+
+	value, err := json.Marshal(update)
+	if err != nil {
+		return err
+	}
+	m.Value = sarama.ByteEncoder(value)
+
+	producer, err := sarama.NewSyncProducer(cfg.Kafka.Addresses, nil)
```

Comment:
We probably don't want to create a new producer for each message.
I'd suggest making a UserUpdateProducer type in the clientapi/producers package along the same lines as the room server producer.